cmd/podman/create.go

```diff
@@ -646,11 +646,6 @@ func parseCreateOpts(ctx context.Context, c *cliconfig.PodmanCommand, runtime *l
 	if util.StringInSlice(".", c.StringSlice("dns-search")) && len(c.StringSlice("dns-search")) > 1 {
 		return nil, errors.Errorf("cannot pass additional search domains when also specifying '.'")
 	}
-	if !netMode.IsPrivate() {
```

Comment:
Do we want to retain this, but change it to only container netmode, so we force DNS sharing if you share netns?

Reply:
I think there is another check that blocks that.  In Docker world it is just ingored.